pyro/infer/reparam/stable.py

```diff
@@ -99,11 +99,92 @@ def __call__(self, name, fn, obs):
                         self._wrap(dist.Exponential(one), event_dim))
 
         # Differentiably transform to scale drawn from a totally-skewed stable variable.
-        z, _ = _unsafe_standard_stable(fn.stability / 2, 1, u, e)
+        a = fn.stability
+        z = _unsafe_standard_stable(a / 2, 1, u, e, coords="S")
         assert (z >= 0).all()
-        scale = fn.scale * (2 ** 0.5) * (math.pi / 4 * fn.stability).cos().pow(1 / fn.stability) * z.sqrt()
+        scale = fn.scale * (math.pi / 4 * a).cos().pow(a.reciprocal()) * z.sqrt()
         scale = scale.clamp(min=torch.finfo(scale.dtype).tiny)
 
-        # Construct a scaled Gaussian.
-        new_fn = self._wrap(dist.Normal(fn.loc, scale), event_dim)
+        # Construct a scaled Gaussian, using Stable(2,0,s,m) == Normal(m,s*sqrt(2)).
+        new_fn = self._wrap(dist.Normal(fn.loc, scale * (2 ** 0.5)), event_dim)
+        return new_fn, obs
+
+
+class StableReparam(Reparam):
+    """
+    Auxiliary variable reparameterizer for arbitrary
+    :class:`~pyro.distributions.Stable` random variables.
+
+    This is useful in inference of non-symmetric
+    :class:`~pyro.distributions.Stable` variables because the
+    :meth:`~pyro.distributions.Stable.log_prob` is not implemented.
+
+    This reparameterizes a :class:`~pyro.distributions.Stable` random variable
+    as sum of two other stable random variables, one symmetric and the other
+    totally skewed (applying Property 2.3.a of [1]). The totally skewed
+    variable is sampled as in :class:`LatentStableReparam` , and the symmetric
+    variable is decomposed as in :class:`SymmetricStableReparam` .
+
+    [1] V. M. Zolotarev (1986)
+        "One-dimensional stable distributions"
+    """
+
+    def __call__(self, name, fn, obs):
+        fn, event_dim = self._unwrap(fn)
+        assert isinstance(fn, dist.Stable)
+
+        # Strategy: Let X ~ S0(a,b,s,m) be the stable variable of interest.
+        # 1. WLOG scale and shift so s=1 and m=0, additionally shifting to convert
+        #    from Zolotarev's S parameterization to Nolan's S0 parameterization.
+        # 2. Decompose X = S + T, where
+        #    S ~ S(a,0,...,0) is symmetric and
+        #    T ~ S(a,sgn(b),...,0) is totally skewed.
```

Comment:
Unless returning a Normal distribution is important, would it be more stable to decompose into 2 totally skewed distributions? (so we have 4 auxiliary random variables instead of 5 as in this PR?)

Reply:
It is indeed important to return a Normal distribution, that way
1. the `.log_prob()` method is define so we can use this as a likelihood, and 
2. we can use this in a `LinearHMM` via `LinearHMMReparam`, where the `Normal` is used as a component distribution in a `GaussianHMM`, permitting parallel-scan inference.

Reply:
Thanks for explaining!